pyproject.toml

```diff
@@ -8,7 +8,7 @@ dev = [
     "mypy>=1.8",  # Pin for mutable-override
     "paper-qa[datasets,ldp,typing,zotero,local]",
     "pre-commit>=3.4",  # Pin to keep recent
-    "pydantic~=2.0,>=2.10.1,<2.10.2",  # Pin 2.10 for typing breaks, downpin 2.10.2 until we fix AgentSettings.callbacks
+    "pydantic~=2.0",
```

Comment:
We still want `>=2.10.1` for the typing breaks

Also, can you propagate to `.pre-commit-config.yaml`